src/callbacks/split_integration.jl

```diff
@@ -0,0 +1,342 @@
+mutable struct SplitIntegrationCallback
+    # Type-stability does not matter here, and it is impossible to implement because
+    # the integration will be created during the initialization.
+    integrator :: Any
+    alg        :: Any
+    kwargs     :: Any
+end
+
+@doc raw"""
+    SplitIntegrationCallback(alg; kwargs...)
+
+Callback to integrate the `TotalLagrangianSPHSystem`s in a `Semidiscretization`
+separately from the other systems.
+After each time step of the main integrator (in which TLSPH systems are ignored),
+the TLSPH systems are integrated for multiple substeps with their own integrator.
+
+This is useful when two conditions are satisfied:
+1. The TLSPH systems require much smaller time steps than the fluid systems,
+   which is usually the case when very stiff materials are simulated.
+2. The amount of TLSPH particles is small compared to the number of fluid particles,
+   so that a fluid time step is much more expensive than a TLSPH substep.
```

Comment:
```suggestion
This is useful under the condition that the TLSPH systems require much smaller time steps than the fluid systems, which is usually the case when stiff materials are simulated.
```

The 2nd case is irrelevant as long as the fluid particles are not much fewer than the TLSPH particles which is not really a realistic case. Even if the TLSPH particles would be the same number as fluid particles the smaller timestep would dominate.

Reply:
Hmmm, not sure. The split integration adds a bit of overhead, so I don't know if it would be noticeably faster if the numbers of fluid and TLSPH particles are the same. Maybe we can rephrase this? It might be useful in this case, but it's the product of the two conditions that determines the speedup factor.

Maybe this?
```suggestion
This is useful if the TLSPH systems require much smaller time steps than the fluid systems,
which is usually the case when stiff materials are simulated.
It is especially useful if additionally the number of TLSPH particles is much smaller than the number of fluid particles.
```